omc3/kmod_averages.py

```diff
@@ -0,0 +1,251 @@
+            tfs.write( opt.output_dir / f'{filename}{EXT}', df, save_index=NAME)
+
+    if opt.plot:
+        plot_kmod_results(
+            data=averaged_results, 
+            ip=opt.ip, 
+            output_dir=opt.output_dir, 
+            show=opt.show_plots
+        )
+
+    averaged_bpm_results[0] = averaged_results
+    return averaged_bpm_results
+
+
+def get_average_betastar_results(meas_paths: Sequence[Path], betastar: float) -> tfs.TfsDataFrame:
+    """
+    Calculate the average betastar results for the given measurements.
+
+    Args:
+        meas_paths: The paths to the measurements.
+        betastar: The model betastar value.
+
+    Returns:
+        The final results as a DataFrame; both beams merged.
+    """
+    final_results = []
+    for beam in [1, 2]:
+        try:
+            all_dfs = [
+                tfs.read(dir_path / f"{BEAM_DIR}{beam}" / f"{RESULTS_FILE_NAME}{EXT}").drop(
+                    columns=[LABEL, TIME]
+                )
+                for dir_path in meas_paths
+            ]
+        except FileNotFoundError as e:
+            LOG.warning(f"Could not find all results for beam {beam}. Skipping.", exc_info=e)
+            continue
+
+        mean_df, std_df =  _get_mean_and_std(all_dfs)
+
+        for column in mean_df.columns:
+            if not column.startswith(ERR):
+                mean_df[f'{ERR}{column}'] = std_df[column]  # TODO: maybe add KMOD errors?
+
+        mean_df[f'{BETASTAR}{MDL}'] = betastar
+        mean_df[BEAM] = beam
+        final_results.append(mean_df)
+    final_results = tfs.concat(final_results).set_index(BEAM)
+    return final_results
+
+
+def get_average_bpm_betas_results(meas_paths: Sequence[Path]) -> dict[int, tfs.TfsDataFrame]:
+    """
+    Calculate the average bpm betas results for the given measurements.
+
+    Args:
+        meas_paths: The paths to the measurements.
+
+    Returns:
+        final_results: A dictionary containing the average bpm betas results for each beam.
+    """
+    final_results = {}
+
+    for beam in [1, 2]:
+        try:
+            all_dfs = [
+                tfs.read(dir_path / f"{BEAM_DIR}{beam}" / f"{LSA_FILE_NAME}{EXT}", index=NAME)
+                for dir_path in meas_paths
+            ]
+        except FileNotFoundError as e:
+            LOG.warning(f"Could not find all results for beam {beam}. Skipping.", exc_info=e)
+            continue
+        
+        mean_df, std_df =  _get_mean_and_std(all_dfs)
+
+        for plane in "XY":
+            mean_df[f'{ERR}{BETA}{plane}'] = std_df[f'{BETA}{plane}']
+        final_results[beam] = mean_df
+    return final_results
+
+
+def _get_mean_and_std(dfs: Sequence[pd.DataFrame]) -> tuple[pd.DataFrame, pd.DataFrame]:  
+    grouped = tfs.concat(dfs, axis=0).groupby(level=0)  # append rows, group by index
+    mean_df = grouped.mean()
+    std_df = grouped.std(ddof=0)  # ddof=0 is default in numpy, ddof=1 is default in pandas
```

Comment:
@fscarlier I think this should be with ddof = 0 only when there is only a single file (to avoid dividing by 0), and ddof = 1 otherwise?  Your numpy implementation from before had ddof=0 by default. Problem is: All the test-files are also now ddof=0 calculated...

Reply:
Sure fine, let's use ddof=1. For single files just check that it is a single file and output that result.